src/app/digital/internal/DigitalComponents.ts

```diff
@@ -93,8 +99,8 @@ export class DigitalComponentInfo implements ComponentInfo {
 const DigitalOutputComponentInfo = (kind: string, outputs: number[], props: TypeMap = {}) =>
     new DigitalComponentInfo(kind, props, { "outputs": "output" }, outputs.map((amt) => ({ "outputs": amt })));
 
-const SwitchInfo = DigitalOutputComponentInfo("Switch", [1]);
-const ButtonInfo = DigitalOutputComponentInfo("Button", [1]);
+const SwitchInfo = DigitalOutputComponentInfo("Switch", [1], { "isOn": "boolean" });
+const ButtonInfo = DigitalOutputComponentInfo("Button", [1], { "isOn": "boolean" });
```

Comment:
Maybe consider a side "propagation-state" that's saved adjacent to the Schema as a "local state" or whatever